sync/sync.go

```diff
@@ -311,30 +311,44 @@ Register:
 	return t.reloadAndVerify(s, replicaAddress, toClient)
 }
 
-func (t *Task) isRevisionCountAndChainSame(fromClient, toClient *replicaClient.ReplicaClient) (bool, error) {
+func (t *Task) isRevisionCountAndChainSame(fromClient, toClient *replicaClient.ReplicaClient) (bool, []string, error) {
 	rwReplica, err := fromClient.GetReplica()
 	if err != nil {
-		return false, err
+		return false, nil, err
 	}
 
 	curReplica, err := toClient.GetReplica()
 	if err != nil {
-		return false, err
+		return false, nil, err
 	}
 	logrus.Infof("RevisionCount of RW replica (%v): %v, Current replica (%v): %v",
 		fromClient.GetAddress(), rwReplica.RevisionCounter, toClient.GetAddress(),
 		curReplica.RevisionCounter)
 	logrus.Infof("RW replica chain: %v, cur replica chain: %v", rwReplica.Chain, curReplica.Chain)
+	if curReplica.Checkpoint != "" {
+		for indx, snapshot := range curReplica.Chain {
+			if snapshot == curReplica.Checkpoint {
+				curReplica.Chain = curReplica.Chain[:indx]
```

Comment:
should the Checkpoint related one also need to be compared?

Reply:
@vishnuitta Not able to grasp this comment

Reply:
Checkpointed snapshot need not be compared as it should be present at all the replicas